docs/development/module/quick.md

```diff
@@ -160,69 +222,186 @@ Available traits and the methods that they provide include:
     given a name and vocabulary. If the term does not exist, a new term will be
     created.
 
-### Dependencies
-
-All dependencies for a quick form should be declared in the module's
-`*.info.yml` file.
-
 ## Quick form actions
 
+farmOS provides lists of logs and assets throughout its interface. Many of
+these lists allow the user to select one or more entities and perform a
+"bulk action" (eg: "Archive asset", "Assign owners", etc).
+
 Quick form actions provide a shortcut to completing a quick form that performs
 actions on or references existing entities.
 
+This allows a user to select one or more entities from a list in farmOS, and be
+redirected to the quick form with the selected entities passed in. These
+selected entities can then be used in the quick form code in various ways.
+
 ### Providing a quick form action
 
-To add a quick form action, a module can provide a PHP class in
-`src/Plugin/Action` that extends the `QuickFormActionBase` class.
+To add a quick form action, two additional files are added to the module:
+
+1. a PHP class in `src/Plugin/Action` that extends from `QuickFormActionBase`
+2. an action config entity in `config/install/system.action.*.yml`
 
-For example, an action to complete the "Egg harvest" quick for select
-assets would be provided as follows:
+For example, an action that redirects to the "Harvest" quick form defined above
+for prepopulating the "Asset" field would be provided as follows:
 
-`src/Plugin/Action/EggHarvest.php`:
+`/src/Plugin/Action/Harvest.php`:
 
 ```php
 <?php
 
-namespace Drupal\farm_egg\Plugin\Action;
+namespace Drupal\farm_quick_harvest\Plugin\Action;
 
 use Drupal\farm_quick\Plugin\Action\QuickFormActionBase;
 
 /**
- * Action for recording egg harvests.
+ * Action for recording harvests.
  *
  * @Action(
- *   id = "egg_harvest",
- *   label = @Translation("Record egg harvest"),
+ *   id = "harvest",
+ *   label = @Translation("Record harvest"),
  *   type = "asset",
- *   confirm_form_route_name = "farm.quick.egg"
+ *   confirm_form_route_name = "farm.quick.harvest"
  * )
  */
-class EggHarvest extends QuickFormActionBase {
+class Harvest extends QuickFormActionBase {
 
   /**
    * {@inheritdoc}
    */
   public function getQuckFormId(): string {
```

Comment:
There's a typo in the method name, the `i` in `Quck` is missing. It is however introduced in the [QuickFromActionBase](https://github.com/farmOS/farmOS/blob/2.x/modules/core/quick/src/Plugin/Action/QuickFormActionBase.php#L73) definition directly so would require fixing the code first, but it might be a nice opportunity to this now while already making changes in quick forms.

Reply:
Oops! Good catch @wotnak! That went unnoticed.

Unfortunately it's a bit tricky to fix because there may already be implementations of this base class in the wild. One that I'm aware of is this: https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/blob/2d8e76f28df8da6dd5a9e108e5bc833e89a5ff2b/modules/farm_rothamsted_quick/src/Plugin/Action/QuickExperimentAction.php#L23

That *may* be the only one (since it was the reason this base class was added originally, so chances are few people know about it), but best practice would be for us to not change it in a way that potentially breaks downstream. I'll chat with @paul121 about it and figure out what we can do.

Thanks for pointing it out!

Reply:
I added a commit to fix this. This IS technically a breaking change, because the method is declared as `abstract` - and there isn't a very easy way to deprecate that and add another, because that would also need to be declared as `abstract`, which would also serve to break existing implementations.

I'm inclined to just fix it, under the assumption that this isn't being used widely downstream yet. I searched ALL code on GitHub (pretty neat that you can do that!), and the only references to it were farmOS core (and forks of it), and the Rothamsted module, which we control: https://github.com/search?q=getQuckFormId&type=code

We'll need to fix it in the Rothamsted module before this gets merged so that doesn't break.

Reply:
Fixed in Rothamsted: https://github.com/Rothamsted-Ecoinformatics/farm_rothamsted/commit/e1e8861776db73fac2b99edac01eb52d698e979c